x-pack/plugins/security_solution/server/lib/detection_engine/signals/bulk_create_threshold_signals.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuidv5 from 'uuid/v5';
+import { reduce, get, isEmpty } from 'lodash/fp';
+import set from 'set-value';
+
+import { Threshold } from '../../../../common/detection_engine/schemas/common/schemas';
+import { Logger } from '../../../../../../../src/core/server';
+import { AlertServices } from '../../../../../alerts/server';
+import { RuleAlertAction } from '../../../../common/detection_engine/types';
+import { RuleTypeParams, RefreshTypes } from '../types';
+import { singleBulkCreate, SingleBulkCreateResponse } from './single_bulk_create';
+import { SignalSearchResponse } from './types';
+
+// used to generate constant Threshold Signals ID when run with the same params
+const NAMESPACE_ID = '0684ec03-7201-4ee0-8ee0-3a3f6b2479b2';
+
+interface BulkCreateThresholdSignalsParams {
+  actions: RuleAlertAction[];
+  someResult: SignalSearchResponse;
+  ruleParams: RuleTypeParams;
+  services: AlertServices;
+  inputIndexPattern: string[];
+  logger: Logger;
+  id: string;
+  filter: unknown;
+  signalsIndex: string;
+  name: string;
+  createdAt: string;
+  createdBy: string;
+  updatedAt: string;
+  updatedBy: string;
+  interval: string;
+  enabled: boolean;
+  refresh: RefreshTypes;
+  tags: string[];
+  throttle: string;
+  startedAt: Date;
+}
+
+interface FilterObject {
+  bool?: {
+    filter?: FilterObject | FilterObject[];
+    should?: Array<Record<string, Record<string, string>>>;
+  };
+}
+
+const getNestedQueryFilters = (filtersObj: FilterObject): Record<string, string> => {
+  if (Array.isArray(filtersObj.bool?.filter)) {
+    return reduce(
+      (acc, filterItem) => {
+        const nestedFilter = getNestedQueryFilters(filterItem);
+
+        if (nestedFilter) {
```

Comment:
I think we can remove the `if` as `getNestedQueryFilters` returns always an empty `{}` and `return { ...acc, ...nestedFilter };` for all cases. What do you think?